tests/conftest.py

```diff
@@ -62,18 +75,405 @@ def conn_params(test_session):
+    return make_run(1, model_1, emission_1)
+
+
+@pytest.fixture(scope='function')
+def run_11(model_1, emission_2):
+    return make_run(2, model_1, emission_2)
+
+
+# DataFile
+
+def make_data_file(i, run=None, timeset=None):
+    return DataFile(
+        id=i,
+        filename='/storage/data_file_{}.nc'.format(i),
+        first_1mib_md5sum='first_1mib_md5sum',
+        unique_id='unique_id_{}'.format(i),
+        x_dim_name='lon',
+        y_dim_name='lat',
+        t_dim_name='time',
+        index_time=datetime.datetime.now(),
+        run=run,
+        timeset=timeset,
+    )
+
+
+@pytest.fixture(scope='function')
+def data_file_1(run_11):
+    return make_data_file(1, run_11)
+
+
+@pytest.fixture(scope='function')
+def data_file_2(run_12):
+    return make_data_file(2, run_12)
+
+
+@pytest.fixture(scope='function')
+def data_file_3(run_11):
+    return make_data_file(3, run_11)
+
+
+# VariableAlias
+
+def make_variable_alias(i):
+    return VariableAlias(
+        long_name='long_name_{}'.format(i),
+        standard_name='standard_name_{}'.format(i),
+        units='units_{}'.format(i),
+    )
+
+
+@pytest.fixture(scope='function')
+def variable_alias_1():
+    return make_variable_alias(1)
+
+
+@pytest.fixture(scope='function')
+def variable_alias_2():
+    return make_variable_alias(2)
+
+
+# DataFileVariableDSGTimeSeries
+
+def make_test_dfv_dsg_time_series(i, file=None, variable_alias=None):
+    return DataFileVariableDSGTimeSeries(
+        id=i,
+        derivation_method='derivation_method_{}'.format(i),
+        variable_cell_methods='variable_cell_methods_{}'.format(i),
+        netcdf_variable_name='var_{}'.format(i),
+        disabled=False,
+        range_min=0,
+        range_max=100,
+        file=file,
+        variable_alias=variable_alias,
+    )
+
+
+@pytest.fixture(scope='function')
+def dfv_dsg_time_series_11(data_file_1, variable_alias_1):
+    return make_test_dfv_dsg_time_series(
+        1, file=data_file_1, variable_alias=variable_alias_1)
+
+
+@pytest.fixture(scope='function')
+def dfv_dsg_time_series_12(data_file_1, variable_alias_2):
+    return make_test_dfv_dsg_time_series(
+        2, file=data_file_1, variable_alias=variable_alias_2)
+
+
+@pytest.fixture(scope='function')
+def dfv_dsg_time_series_21(data_file_2, variable_alias_1):
+    return make_test_dfv_dsg_time_series(
+        3, file=data_file_2, variable_alias=variable_alias_1)
+
+
+@pytest.fixture(scope='function')
+def dfv_dsg_time_series_31(data_file_3, variable_alias_1):
+    return make_test_dfv_dsg_time_series(
+        4, file=data_file_3, variable_alias=variable_alias_1)
+
+
+# Ensemble
+
+def make_ensemble(id):
+    return Ensemble(
+        id=id,
+        changes='wonder what this is for',
+        description='Ensemble {}'.format(id),
+        name='ensemble_{}'.format(id),
+        version=float(id)
+    )
+
+
+@pytest.fixture(scope='function')
+def ensemble_1():
+    return make_ensemble(1)
+
+
+@pytest.fixture(scope='function')
```

Comment:
Much of the fixture creation code here is pretty verbose. Are there any alternate methods for registering `pytest` fixtures, such that you could parameterize it, or do a handful of them in one shot?

Reply:
That thought crossed my mind repeatedly. I will explore a couple of my ideas, which fall into the following categories:
- I don't think pytest directly provides any alternatives for registering fixtures, but I will investigate.
- Metaprogramming? Not feeling extremely confident about this on first pass.
- Instead of individual fixtures per object, provide one fixture that returns a list of objects, and select from them, e.g., `def test(things): assert things[0].name == thing0`. A bit nasty, but not really much different from individual fixtures named `thing0`, `thing1` as we have now.
- Indirect fixtures: 
   - For a given set of `Thing` objects, the fixture `things` is used [indirectly](https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments): it is passed a value from the test parametrization. These  
   - This also solves the issue of parametrizing over objects, which is not possible without doing this, and for which there are several tests that would be much better with this facility.

Unsurprisingly, I am favouring indirect fixtures, even though they complicate things a bit ... in order to simplify them.

Reply:
1. Apparently not.
2. Or on second pass.

Reply:
I implemented much more compact, readable, and largely self-documenting way of creating the test objects, and provide them through parametrized fixtures, which also improves several of the tests. Unfortunately you can't register pytest fixtures in a loop (which would have been very convenient here), but the repetition is still much reduced.